.yamllint.conf

```diff
@@ -1,9 +1,29 @@
-
+---
+# .yamllint.conf
 extends: default
 
 rules:
+  # General rules
   line-length:
-    max: 100
+    max: 100  # Aligns with your preference for a maximum line length of 100
     level: warning
+    allow-non-breakable-words: true
   indentation:
-    indent-sequences: whatever
\ No newline at end of file
+    spaces: 2  # Set indentation to 2 spaces for YAML files
+    indent-sequences: whatever
+  hyphens:
+    max-spaces-after: 1  # Enforce spacing after hyphens in lists
+
+  # Comments
+  comments:
+    min-spaces-from-content: 2
+
+ignore: |
+  node_modules/**
+  dist/**
+  build/**
+  .git/**
+  venv/**
+  __pycache__/**
+  *.xc**
+  .travis.yml
```

Comment:
🙅🏻 for some reason there is a bug with the version of `yamllint` used in CI where it can not correctly parse our copy of `.travis.yml` so this file needs to be ignored in our case. In general you'd have been right, this is just a special-case.